app/controllers/api/v1/users_controller.rb

```diff
@@ -56,23 +56,24 @@ def purge_avatar
         render_json status: :ok
       end
 
-      # POST /api/v1/users/:id/change_password.json
+      # POST /api/v1/users/change_password.json
       # Expects: { user: { :old, :new } }
       # Returns: { data: Array[serializable objects] , errors: Array[String] }
       # Does: Validates and change the user password.
 
       def change_password
-        # TODO:  add a before_action callback to find_user.
-        user = User.find(params[:id])
+        return render_json status: :unauthorized unless current_user # TODO: generalise this.
```

Comment:
Admins should be able to change a user's password, so we shouldn't switch this to current_user

Reply:
An admin can change a user password?
The `change_password` expects the current password and a new one.
If the admin can change any user password they need to provide their current password which is odd.
Or, Admins can reset any user password without any need to provide their current credentials.
You mean admins can reset user's password?
If that's the case isn't that a separate endpoint other then this one?

Reply:
Yeah sorry I mixed them up. An admin can send a reset password email